.github/workflows/version-check.yml

```diff
@@ -15,16 +15,27 @@ jobs:
         run: |
           GRADLE_VERSION=$(awk -F= 'NR==1 {print $2}' ./service/gradle.properties)
           NPM_VERSION=$(jq -r '.version' ./client/package.json)
+
+          echo "Gradle version: '$GRADLE_VERSION'."
+          echo "NPM version:    '$NPM_VERSION'."
+
           if [ "$GRADLE_VERSION" != "$NPM_VERSION" ]; then
             echo "Version mismatch between client and service. Update the version in both modules."
             exit 1
           fi
 
       - name: Check version equality between branches
         run: |
+          WORK_VERSION=$(awk -F= 'NR==1 {print $2}' ./service/gradle.properties)
+
           git fetch origin main
+          git checkout main
```

Comment:
that checkout feels strange, but If it is in this workflow context then probably that's not a problem